common/utils.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any
+from dataclasses import dataclass
+from pathlib import Path
+
+from common.consts import MINIO_MAIN_PATH, PROJECT_PATH
+from metrics.consts import MetricCollections
+
+
+def singleton(cls) -> Any:
+    """Singleton implementation in form of decorator"""
+    instances = {}
+
+    def getinstance() -> Any:
+        if cls not in instances:
+            instances[cls] = cls()
+        return instances[cls]
+
+    return getinstance
+
+
+def create_weights_path(prefix: Path, collection_name: MetricCollections, suffix: str) -> Path:
+    return prefix / "data" / "models" / collection_name.value / suffix
+
+
+@dataclass
+class WeightsData:
+    """Dto containing all dataset information"""
+    collection_name: MetricCollections
+    trunk_minio: Path = MINIO_MAIN_PATH
+    embedder_minio: Path = MINIO_MAIN_PATH
+    trunk_local: Path = PROJECT_PATH
+    embedder_local: Path = PROJECT_PATH
+
+    def __post_init__(self) -> None:
+        self.trunk_minio = create_weights_path(
```

Comment:
you can use dataclass.field for defininig default values by providing factory as lambda func

Reply:
Also what is the purpose of specific dto when everything seems constant?